pkg/api/api.go

```diff
@@ -103,6 +103,26 @@ type IssueCodeResponse struct {
 	ErrorCode          string `json:"errorCode,omitempty"`
 }
 
+// CheckCodeStatusRequest defines the parameters to request the status for a
+// previously issued OTP code. This is called by the Web frontend.
+// API is served at /api/codestatus
+type CheckCodeStatusRequest struct {
+	ID []string `json:"id"`
+}
+
+// CheckCodeStatusResponse defines the response type for CheckCodeStatusRequest.
+type CheckCodeStatusResponse struct {
+	CodeStatus map[string]CodeStatus `json:"codeStatus"`
+}
+
+// CodeStatus defines the status of an OTP code.
+type CodeStatus struct {
+	Claimed            bool   `json:"claimed"`
+	VerificationCode   string `json:"code"`
```

Comment:
not sure if we want to repeat the actual verification code

Reply:
apologies. I intended to include the ID, not the code. That said what is the risk of repeating the code?